Game4.py

```diff
@@ -20,73 +33,96 @@ def __init__(self, width=480,height=360):
         # create clock
         self.clock = pygame.time.Clock()
         self.score = 0
+        self.oneplayer = True
 
 
-    def mainLoop(self):
+    def start_menu(self):
+        """Start screen for game."""
+        while True:
+            if pygame.QUIT in {event.type for event in pygame.event.get()}:
+                sys.exit()
+
+            # display start menu
+            self.screen.fill(colors['BLACK'])
+            # title
+            font = pygame.font.SysFont("comicsansms", int(self.width/8))
+            text = font.render("Blockade", True, colors['WHITE'])
+            text_rect = text.get_rect(center=(self.width/2, self.height/3))
+            self.screen.blit(text, text_rect)
+            # instructions
+            font = pygame.font.SysFont("comicsansms", int(self.width/30))
+            text = font.render("Player 1: Arrow Keys, Player 2: ZXC", True, colors['WHITE'])
+            text_rect = text.get_rect(center=(self.width/2, self.height/2))
+            self.screen.blit(text, text_rect)
+            # buttons
+            self.button("One Player",self.width*1/8-20,self.height*3/4,colors['GREEN'],True,self.main_loop)
+            self.button("Two Player",self.width*5/8-20,self.height*3/4,colors['RED'],False,self.main_loop)
+
+            # update screen
+            pygame.display.flip()
+
+
+    def main_loop(self):
         """Main screen for game."""
-        # initialize player
+        # start count
         count = 0
-        play_len = 25
-        play_x = 100
-        play_y = self.height - play_len
-        player = Player(play_x, play_y, play_len, self.screen)
+
+        # initialize player
+        player = Player(PLAY_X, self.height-PLAY_LEN, PLAY_LEN, self.screen)
 
         # initialize stamina bars
-        P1_stamina_bar = StaminaBar(self.screen,25,"WHITE")
-        P2_stamina_bar = StaminaBar(self.screen,350,"RED")
+        P1_stamina_bar = StaminaBar(self.screen, P1_STAMINA_BAR_OFFSET, 'WHITE')
+        P2_stamina_bar = StaminaBar(self.screen, self.width - P2_STAMINA_BAR_OFFSET, 'RED')
 
         # initialize obstacle length, x and y coordinates
-        obs_len = 25
-        obs_x = self.width
-        obs_y = self.height - obs_len
-        obstical_height = self.height - obs_len
+        OBS_X = self.width
+        OBS_Y = self.height - OBS_LEN
 
         # create list of obstacles
-        obstacles = []
-        new_obstical = False
-        # uncomment and place line below in [] of obstacles = [] to generate random obstacles:
-        # Obstacle(obs_x, obs_y, obs_len, self.screen,'BLUE')
-
+        if self.oneplayer:
+            obstacles = [Obstacle(OBS_X, OBS_Y, OBS_LEN, self.screen,'BLUE')]
+        else:
+            obstacles = []
+            new_obstacle = False
+            obstacle_height = OBS_Y
 
         # initialize time variables
         prev_time = 0
         obs_dt = 500
 
         # main event loop
-        while 1:
+        while True:
```

Comment:
okay while `True` is better than while `1` but it still does not tell anything about the state of the while loop. I suggest using `while running = True` or `while playing`